framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -50,31 +61,114 @@ class MeshGenerator : public MooseObject, public MeshMetaDataInterface
    * Internal generation method - this is what is actually called
    * within MooseApp to execute the MeshGenerator.
    */
-  std::unique_ptr<MeshBase> generateInternal();
+  [[nodiscard]] std::unique_ptr<MeshBase> generateInternal();
+
+  /**
+   * @returns The names of the MeshGenerators that were requested in the getMesh methods
+   */
+  const std::set<MeshGeneratorName> & getRequestedMeshGenerators() const
+  {
+    return _requested_mesh_generators;
+  }
+  /**
+   * @returns The names of the MeshGenerators that were requested in the getMeshForSub methods
+   */
+  const std::set<MeshGeneratorName> & getRequestedMeshGeneratorsForSub() const
+  {
+    return _requested_mesh_generators_for_sub;
+  }
+
+  /**
+   * Class that is used as a parameter to add[Parent/Child]() that allows only
+   * MooseApp methods to call said methods
```

Comment:
Maybe mention that this is called the "Passkey pattern. IMO using the generic templated version would be neat here.

```
template <typename T>
class Key 
{ 
  friend T;
  Key() {}
  Key(Key const&) = default; 
};
```

That make it obvious in the function signature which class is permitted to call the method. I suppose this would not work for the general case where multiple classes are permitted to call, but maybe there is some variadic template magic to generalize this?

Reply:
I like this. Looks pretty.

Reply:
lol. Death by C++ incoming

Reply:
This is already an obscure idiom, templating it does not help

Reply:
@dschwen said he would use the PassKey idiom in one of his future PR (much to my dismay but I think we agree to disagree) so he may address both his and my comment at this time.